easybuild/easyconfigs/o/OTF2/OTF2-3.0.3-GCCcore-13.3.0.eb

```diff
@@ -0,0 +1,55 @@
+# #
+# This is an easyconfig file for EasyBuild, see https://github.com/easybuilders/easybuild
+# Copyright:: Copyright 2013-2024 Juelich Supercomputing Centre, Germany
+# Authors::   Bernd Mohr <b.mohr@fz-juelich.de>
+# Markus Geimer <m.geimer@fz-juelich.de>
+# Jan André Reuter <j.reuter@fz-juelich.de>
+# License::   3-clause BSD
+#
+# This work is based on experiences from the UNITE project
+# http://apps.fz-juelich.de/unite/
+# #
+
+easyblock = 'EB_Score_minus_P'
+
+name = 'OTF2'
+version = '3.0.3'
+
+homepage = 'https://www.score-p.org'
+description = """
+ The Open Trace Format 2 is a highly scalable, memory efficient event trace
+ data format plus support library. It is the new standard trace format for
+ Scalasca, Vampir, and TAU and is open for other tools.
+
+"""
+
+toolchain = {'name': 'GCCcore', 'version': '13.3.0'}
+
+source_urls = ['http://perftools.pages.jsc.fz-juelich.de/cicd/%(namelower)s/tags/%(namelower)s-%(version)s']
+sources = [SOURCELOWER_TAR_GZ]
+patches = ['otf2-3.0.3-py-compile-python-3.12-imp-removal.patch']
+checksums = [
+    '18a3905f7917340387e3edc8e5766f31ab1af41f4ecc5665da6c769ca21c4ee8', # otf2-3.0.3.tar.gz
+    '95cf3c4fcf5bc9785eb92433fa28631d3f7ebce4b9d9e4871f67c5424f677156', # otf2-3.0.3-py-compile-python-3.12-imp-removal.patch
```

Comment:
This patch might be required for other OTF2 versions as well. This seems to depend on the system Python version (or `EB_PYTHON`). Do we want to add the patch for all the OTF2 3.0.3 versions? If so, do we want to do this in this PR or a separate one?

Reply:
Looking into it a bit more, should OTF2 even use the system Python at all?

```
Configure command:
  ./configure                   '--prefix=/data/EasyBuild-develop/software/OTF2/3.0.3-GCCcore-13.3.0' \
                                '--build=x86_64-pc-linux-gnu' \
                                '--host=x86_64-pc-linux-gnu' \
                                '--enable-shared' \
                                '--with-nocross-compiler-suite=gcc' \
                                '--with-libbfd-include=/data/EasyBuild-develop/software/binutils/2.42-GCCcore-13.3.0/include' \
                                '--with-libbfd-lib=/data/EasyBuild-develop/software/binutils/2.42-GCCcore-13.3.0/lib' \
                                '--with-sionlib=/data/EasyBuild-develop/software/SIONlib/1.7.7-GCCcore-13.3.0-tools/bin' \
                                'build_alias=x86_64-pc-linux-gnu' \
                                'host_alias=x86_64-pc-linux-gnu'

Loaded modules:
  module load                   GCCcore/13.3.0 \
                                zlib/1.3.1-GCCcore-13.3.0 \
                                binutils/2.42-GCCcore-13.3.0 \
                                SIONlib/1.7.7-GCCcore-13.3.0-tools

Configuration summary:
  OTF2 3.0.3:
    Platform:                   linux (auto detected)
    Cross compiling:            no (auto detected)

    OTF2 (backend):
      C99 compiler:             /data/EasyBuild-develop/software/GCCcore/13.3.0/bin/gcc (gnu 13.3.0)
      Link mode:                static=yes, shared=yes
      SIONlib support:          yes, using /data/EasyBuild-develop/software/SIONlib/1.7.7-GCCcore-13.3.0-tools/bin/sionconfig --be \
                                --gen --threadsafe --64
      GCC atomic builtins:      yes
      TLS support:              yes, using __thread and the initial-exec model
      Entropy sources:          clock_gettime getpid sysinfo gethostid
      Python bindings support:  yes, using /usr/bin/python3
      Running tests:            no
        Parallel tests:         yes
```

The correct way would probably be to either disable `Python bindings support` altogether or use a built Python module instead, right?

Reply:
"Python bindings support" looks like it installs some python package and hence should have a dependency on a Python EC

Reply:
I opened https://github.com/easybuilders/easybuild-easyconfigs/issues/21266 for this and will look into it after both Score-P/Scalasca related PRs are merged, just to make sure that there are not too many changes at similar files at the same time.